### ISSUE TITLE
Why did you transposed twice in contextual_attention

### ISSUE THREAD

**Issue by dayeguilaiye** [JiahuiYu/generative_inpainting#272]
Hi, I found that you  have transposed twice in contextual_attention layer, but I can't understand why. Could you tell me the reason?

Thanks a lot !!!

```python
if fuse:
            yi = tf.reshape(yi, [1, fs[1]*fs[2], bs[1]*bs[2], 1])
            yi = tf.nn.conv2d(yi, fuse_weight, strides=[1,1,1,1], padding='SAME')
            yi = tf.reshape(yi, [1, fs[1], fs[2], bs[1], bs[2]])
            yi = tf.transpose(yi, [0, 2, 1, 4, 3])
            yi = tf.reshape(yi, [1, fs[1]*fs[2], bs[1]*bs[2], 1])
            yi = tf.nn.conv2d(yi, fuse_weight, strides=[1,1,1,1], padding='SAME')
            yi = tf.reshape(yi, [1, fs[2], fs[1], bs[2], bs[1]])
            yi = tf.transpose(yi, [0, 2, 1, 4, 3])
yi = tf.reshape(yi, [1, fs[1], fs[2], bs[1]*bs[2]])
```

**Comment by dayeguilaiye** [comment]
And I don't understand why did you do the " / 4." here:
`yi = tf.nn.conv2d_transpose(yi, wi_center, tf.concat([[1], raw_fs[1:]], axis=0), strides=[1,rate,rate,1]) / 4.`
in "inpaint_ops.py", too

**Comment by JiahuiYu** [comment]
@dayeguilaiye Hi, `/ 4` is for averaging pixel values. `Concat` is for batching.

The implementation of contextual attention using tensorflow is a bit difficult to understand.
